ui/src/pages/accounts/Login/Login.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { LoginWithOauthButton } from '@authentication/components';
+import { Grid, Header, Divider, Segment } from 'semantic-ui-react';
+
+import { LoginWithLocalAccountForm } from './components';
+import { parse } from '../utils';
+import { FrontSiteRoutes } from '@routes/urls';
+import { goTo } from '@history';
+import {
+  OAUTH_PROVIDERS,
+  ENABLE_LOCAL_ACCOUNT_LOGIN,
+  ENABLE_OAUTH_LOGIN,
+} from '@config';
+
+export default class Login extends Component {
+  componentDidMount() {
+    const params = parse(window.location.search);
+    if ('sessionExpired' in params) {
+      this.props.sendErrorNotification(
+        'Session Error',
+        'You are either not signed in or your session has expired.'
+      );
+    }
+  }
+
+  redirectIfAlreadyLoggedIn = params => {
+    if (!this.props.isLoading && !this.props.isAnonymous) {
+      if (!('sessionExpired' in params)) {
+        this.props.clearNotifications();
+        goTo(params.next || FrontSiteRoutes.home);
+      }
+    }
+  };
+
+  checkIfOauthLoginResponse = params => {
+    if (params.code) {
```

Comment:
These can be combine in one condition 
`if (params.code && params.code === 200) ...`